backends/p4tools/README.md

```diff
@@ -0,0 +1,110 @@
+# P4Tools - Testing Tools For P4 Targets
+
+## Directory Structure
+
+```
+p4tools
+ ├─ cmake       ── common CMake modules
+ ├─ common      ── C++ source: common code for the various components of p4check
+ ├─ mutate      ── C++ source: p4mutate
+ ├─ p4check     ── C++ source: the main entry point for p4check
+ ├─ scripts
+ │   ├─ hooks   ── git hooks
+ │   └─ tools   ── development-support scripts (e.g., cpplint)
```

Comment:
can't we reuse the existing cpplint tool?

Reply:
Yes, I would do this in a separate PR. We can move up a lot of the tools here, they are generally useful.